mods/default/functions.lua

```diff
@@ -366,8 +364,7 @@ minetest.register_abm({
 		end
 
 		-- Look for likely neighbors.
-		local p2 = minetest.find_node_near(pos, 1, {"default:dirt_with_grass",
-				"default:dirt_with_dry_grass", "default:dirt_with_snow"})
+		local p2 = minetest.find_node_near(pos, 1, "group:spreading_dirt_type")
 		if p2 then
 			-- But the node needs to be under air in this case.
 			local n2 = minetest.get_node(above)
```

Comment:
Well, it's highly likely the dirt will be under air as it has 'neighbors = air, grasses, snow'. But your suggestion will simplify the code.